src/library/searchquery.cpp

```diff
@@ -166,6 +166,26 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+bool NullTextFilterNode::match(const TrackPointer& pTrack) const {
+    for (const auto& sqlColumn: m_sqlColumns) {
+        QVariant value = getTrackValueForColumn(pTrack, sqlColumn);
+        if (!value.isValid() || !qVariantCanConvert<QString>(value)) {
```

Comment:
This condition might cause that different columns are picked by match() and toSql(). toSql() simply picks the first column.

Reply:
OK. This is only relevant for "artist" which has two columns. So I think in our case if the "artist" column is empty or invalid it should be enough to match.